gbn/gbn_conn.go

```diff
@@ -374,7 +376,25 @@ func (g *GoBackNConn) sendPacket(ctx context.Context, msg Message) error {
 func (g *GoBackNConn) sendPacketsForever() error {
 	// resendQueue re-sends the current contents of the queue.
 	resendQueue := func() error {
-		return g.sendQueue.resend()
+		err := g.sendQueue.resend()
```

Comment:
Hmmm sorry I don't fully understand what this is suggesting that I should do? Would you like me to include a cleanup in the `resendQueue` func here if it the `err := g.sendQueue.resend()` yields an error?

Reply:
im just saying that it is strange to not check the error immediately after calling g.sendQueue.resend(). 

So it is weird to have this pattern:

```
err = somethingThatCouldError()

<other logic> 

if err != nil {
    <do thing with error> 
}
```

It's better to have the structure of:

```
err = somethingThatCouldError()
if err != nil {
    <do thing with error> 
}

<other logic> 
```

The difference between A & B here though is that the <other logic> always gets called in A whereas it wont necessarily get called in B. So im saying that if we go with format B and you want to make sure that the <other logic> gets called - just add that other logic here:

```
err := g.sendPacketsForever()
if err != nil {
        <other logic> 
}
```

since that will get called if we do get an error on `g.sendQueue.resend()`

does that make sense?

Reply:
Ah, now I understand what you were referring to, thanks!

I changed this to now return the error immediately after `g.sendQueue.resend()` has been executed if it errors.

It wasn't intentional previously that rest of the code in `resendQueue` _MUST_ run even if `g.sendQueue.resend()` had errored, it was more a case of that it didn't matter wether it did or not and it then previously skipped adding an extra if case. But I understand now why this is bad code design, thanks :)!